web/terraform-checkpoints/create-firebase-project-checkpoint.tf

```diff
@@ -0,0 +1,69 @@
+# Terraform configuration to set up providers by version.
+terraform {
+  required_providers {
+    google-beta = {
+      source  = "hashicorp/google-beta"
+      version = "~> 4.0"
+    }
+  }
+}
+
+# Configure the provider not to use the specified project for quota check.
+# This provider should only be used during project creation and initializing services.
+provider "google-beta" {
+  alias                 = "no_user_project_override"
+  user_project_override = false
+}
+
+# Configure the provider that uses the new project's quota.
+provider "google-beta" {
+  user_project_override = true
+}
+
+# Create a new Google Cloud project.
+resource "google_project" "default" {
+  provider = google-beta.no_user_project_override
+
+  name       = "<PROJECT_NAME_OF_YOUR_PROJECT>"
```

Comment:
What do you think about adding a comment (here or at the top of the resource) to highlight these parameters that need to be replaced and their targetted values?

Reply:
I would also write these TODOs in all-caps.

Reply:
Done.